File: 0290-word-pattern/0290-word-pattern.java
class Solution {
    public boolean wordPattern(String pattern, String s) {
        String[] ar = s.split(" ");
        if(ar.length!=pattern.length())return false;
        HashMap<String,String> hm = new HashMap<>();
        for(int i=0;i<pattern.length();i++){
            String temp =pattern.charAt(i)+"";
            if(hm.containsKey(temp)&&!hm.get(temp).equals(ar[i]))return false;
            else if(!hm.containsKey(temp)&&hm.containsValue(ar[i]))return false;
            else hm.put(temp,ar[i]);
        }
        return true;
    }
}

File: 0290-word-pattern/0290-word-pattern.php
class Solution {

    /**
     * @param String $pattern
     * @param String $s
     * @return Boolean
     */
    function wordPattern($pattern, $s) {
        $ar=explode(" ",$s);
        $len=count($ar);
        $ar_check=array();
        if(strlen($pattern)!=$len)return false;
        for($i=0;$i<count($ar);$i++){
           if(array_key_exists($pattern[$i],$ar_check)&&$ar_check[$pattern[$i]]!=$ar[$i])return false;
            else $ar_check[$pattern[$i]]=$ar[$i];
        }
         $counts = array_count_values($ar_check);
        foreach ($counts as $count) {
            if ($count > 1) {
                return false;
            }
        }
        return true;
    }
}